frontend/src/app/AdminNavBar/AdminNavBar.tsx

```diff
@@ -78,13 +73,40 @@ export interface AdminNavBarProps {
 }
 
 export const AdminNavBar = ({ isMenuOpen }: AdminNavBarProps): JSX.Element => {
-  const { user } = useUser()
+  const { user, isLoading: isUserLoading } = useUser()
 
-  const {
-    isOpen: isContactModalOpen,
-    onClose: onContactModalClose,
-    onOpen: onContactModalOpen,
-  } = useDisclosure()
+  const emergencyContactKey = useMemo(
+    () => (user?._id ? EMERGENCY_CONTACT_KEY_PREFIX + user._id : null),
+    [user],
+  )
+
+  const [hasSeenContactModal, setHasSeenContactModal] =
+    useLocalStorage<boolean>(emergencyContactKey)
+
+  const [isContactModalOpen, setIsContactModalOpen] = useState<boolean>(false)
+  const handleContactModalOpenClick = () => {
+    setIsContactModalOpen(true)
+  }
+  const onEmergencyContactModalClose = () => {
+    setIsContactModalOpen(false)
+    setHasSeenContactModal(true)
+  }
+
+  const isEmergencyContactModalOpenOnLogin = useMemo(
+    () => !isUserLoading && !hasSeenContactModal && !user?.contact,
+    [isUserLoading, hasSeenContactModal, user],
+  )
+
+  const isEmergencyContactModalOpen =
+    (isContactModalOpen && Boolean(user)) || isEmergencyContactModalOpenOnLogin
+
+  const handleLogout = () => {
+    logout()
+    if (emergencyContactKey) {
+      localStorage.removeItem(emergencyContactKey)
+    }
+    window.location.reload()
```

Comment:
Do you want to double check if this is required? Logout clears the logged in storage and should automatically change auth state (meaning protected routes will redirect to login page)

Reply:
as discussed, I've removed `window.location.reload()` from here, and also now remove `LOGGED_IN_KEY` from localStorage in  `AuthService`'s `logout`